ivadomed/utils.py

```diff
@@ -620,6 +640,115 @@ def threshold_predictions(predictions, thr=0.5):
     return thresholded_preds
 
 
+def segment_volume(fname_model, fname_model_metadata, fname_image, fname_roi=None):
+    """Segment an image.
+
+    Segment an image (fname_image) using a already trained model (fname_model) given its
+    training parameters (fname_model_metadata). If provided, a RegionOfInterest (fname_roi)
+    is used to crop the image prior to segment it.
+
+    Args:
+        fname_model (string): model filename (.pt) to use.
+        fname_model_metadata (string): Configuration file filename (.json) used for the training,
+            see https://github.com/neuropoly/ivado-medical-imaging/wiki/configuration-file
+        fname_image (string): image filename (e.g. .nii.gz) to segment.
+        roi_fname (string): Binary image filename (e.g. .nii.gz) defining a region of interest,
+            e.g. spinal cord centerline, used to crop the image prior to segment it if provided.
+    Returns:
+        nibabelObject: Object containing the soft segmentation.
+
+    """
+    # Define device
+    device = torch.device("cpu")
```

Comment:
In some cases, it might be good to benefit from GPU computation, for example for large volume at high resolution ?

Reply:
Agreed it would be a nice feature especially for batch processing, but it opens up lots of potential issues / incompatibilities on the user side (CUDA version, nvidia driver version, pytorch version)

Reply:
That's without considering platform issues, I guess GPUs would be Linux only? Pytorch binaries don't support CUDA on Mac and the same goes for WSL https://github.com/microsoft/WSL/issues/829

Reply:
Thanks for your opinion --> I would suggest to keep cpu only for now then, while keeping this option in mind for the future.